frontend/src/app/commonComponents/Header.jsx

```diff
@@ -1,13 +1,66 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
 import PropTypes from "prop-types";
 import { Link, NavLink } from "react-router-dom";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
 import { PATIENT_TERM_PLURAL_CAP } from "../../config/constants";
 import classNames from "classnames";
-import iconClose from "../../img/close-gray-60.svg";
+import { gql, useQuery } from "@apollo/client";
+import Button from "./Button";
+import { getStaffColor } from "../utils";
+
+const WHOAMI_QUERY = gql`
+  {
+    whoami {
+      id
+      firstName
+      middleName
+      lastName
+      suffix
+      organization {
+        testingFacilityName
+      }
+    }
+  }
+`;
 
 const Header = ({ organizationId }) => {
   const [menuVisible, setMenuVisible] = useState(false);
+  const [staffDetailsVisible, setUserDetailsVisible] = useState(false);
+  const { data: whoamidata } = useQuery(WHOAMI_QUERY, {
+    fetchPolicy: "no-cache",
+  });
+  const [staffInitials, setStaffInitials] = useState("");
+  const [staffName, setStaffName] = useState("");
+  const [facilityName, setFacilityName] = useState("");
+  const [staffIconColor, setStaffIconColor] = useState("");
+  useEffect(() => {
+    if (!whoamidata || !whoamidata.whoami) return;
+    const whoami = whoamidata.whoami;
+    setStaffName(formatFullName(whoami));
+
+    const ch1 = whoami.firstName ? whoami.firstName.charAt(0) : "";
+    const ch2 = whoami.middleName ? whoami.middleName.charAt(0) : "";
+    const ch3 = whoami.lastName ? whoami.lastName.charAt(0) : "";
+    setStaffInitials(`${ch1}${ch2}${ch3}`);
+
+    setFacilityName(whoami.organization.testingFacilityName);
+    setStaffIconColor(getStaffColor(whoami.id));
+  }, [staffIconColor, whoamidata]);
+
+  const formatFullName = (whoami) => {
+    // this trick will not include spaces if middlename is blank.
+    let result = whoami.firstName;
+    result += whoami.middleName ? ` ${whoami.middleName}` : "";
+    result += whoami.lastName ? ` ${whoami.lastName}` : "";
+    result += whoami.suffix ? `, ${whoami.suffix}` : "";
+    return result;
+  };
+
+  const logout = () => {
+    // todo: do logout here. maybe https://developer.okta.com/docs/guides/sign-users-out/react/sign-out-of-your-app/
+    debugger;
+    alert("OKTA logout here");
```

Comment:
If it's okay to not have logout for MVP we could ship after removing these two lines. Another half-option would be to just redirect the user to the home page without logging them out.

Reply:
I'm waiting to hear back from @nickrobison-usds about the link to use. I think @TimBest wants to hold off on the commit until it's in the change.

Reply:
Happy to ship the changes with the logout as `display:none` it you want

Reply:
Let's talk about this at standup? I feel like this whole feature is more about getting usability feedback from users than actual functionality at this stage.